src/Workspaces/Core/Portable/Shared/Extensions/SemanticModelExtensions.cs

```diff
@@ -256,9 +257,16 @@ public static SemanticModel GetOriginalSemanticModel(this SemanticModel semantic
             {
                 if (child.IsNode)
                 {
-                    GetAllDeclaredSymbols(semanticModel, child.AsNode(), symbols, cancellationToken);
+                    var childNode = child.AsNode();
+                    if (ApplyFilter(childNode, filter))
```

Comment:
ah, based on thsi, we should really call the variable `descendInto`.  I would also update the comment on the first file to say that we basically don't want to dive into a local function as they are not affected by outer variables, and do not contribute to the outer scope as well.

Reply:
Yep, I couldn't think of a better name at the time, but ```descendInto``` is good.  Will update!